grafana/grafana
# Influxdb: Remove backend migration feature toggle

**What is this feature?**

This PR;
- Removes the redundant feature toggle we have `influxdbBackendMigration` 
- Removes the access mode checks
- Removes lines related to browser access mode

**Why do we need this feature?**

We migrated influxdb to server access mode (https://github.com/grafana/grafana/issues/43076) only and removed browser access. But there were some lines still checking the access mode. So I cleaned up the code and removed the feature toggle.

**Who is this feature for?**

People who use influxdb

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/60321

**Special notes for your reviewer**:
- I add `breaking change` because levitate needs it. It is a breaking change because we remove the feature toggle that has been useless for a while. So it is not a real breaking change but still.

**How to test**:
There are an onerous steps for testing.
- run `make devenv sources=influxdb,influxdb1` in grafana root on main branch
- Have Grafana v8.4.4 (may need to use `yarn start --env noTsCheck=1`)
- Set up Influxdb 1.8 with InfluxQL with browser access mode
  - CORS will be a problem so create a proxy server in between grafana and influxdb via `express`
  - Permit the `localhost`. Example code is below.
  - Use `localhost:8888` as datasource URL.
  - User `localhost:8087` as targetURL in your proxy server
  - Datasource settings:
    - `make devenv sources=influxdb1,influxdb`
    - Use with auth: username: grafana password: grafana  
    - Use `site` as database in the InfluxDB Details
- Create your dashboard and save it.
- Switch back to this branch
- Update your datasource config because it simply warns you to use `server access mode`
- No need express proxy server so use influxdb URL directly.
  - change url in config options to `localhost:8087`
- Open your test dashboard and see everything is working as it should be.


```js
// This is a simple express proxy server that runs on `:8888`
var express = require('express'),
    cors = require('cors'),
    request = require('request'),
    bodyParser = require('body-parser'),
    app = express();

var myLimit = typeof(process.argv[2]) != 'undefined' ? process.argv[2] : '100kb';
console.log('Using limit: ', myLimit);

const whitelist = ['http://localhost:3000','http://localhost:3001'];
const corsOptions = {
  credentials: true, // This is important.
  origin: (origin, callback) => {
    if(whitelist.includes(origin))
      return callback(null, true)

      callback(new Error('Not allowed by CORS'));
  }
}

app.use(cors(corsOptions));

app.use(bodyParser.json({limit: myLimit}));

app.all('*', function (req, res, next) {

    if (req.method === 'OPTIONS') {
        console.log("CORS Preflight")
        // CORS Preflight
        res.send();
    } else {
        var targetURL = req.header('Target-URL'); // Target-URL ie. https://example.com or http://example.com
        if (!targetURL) {
            // I use the commented lines below for cURL testing
            // res.send(500, { error: 'There is no Target-Endpoint header in the request' });
            // return;
            targetURL = "http://localhost:8087" // Your influxdb url
        }
        request({ url: targetURL + req.url, method: req.method, json: req.body, headers: {'Authorization': req.header('Authorization')} },
            function (error, response, body) {
                if (error) {
                    console.error('error: ' + response.statusCode)
                }
               console.log(body);
            }).pipe(res);
    }
});

app.set('port', process.env.PORT || 8888);

app.listen(app.get('port'), function () {
    console.log('Proxy server listening on port ' + app.get('port'));
});
```


# Release notice breaking change

Removes the non-functional feature toggle `influxdbBackendMigration`. InfluxDB is working %100 with server access mode. You can keep using your dashboards, and data sources as you have been using. This won't affect them. If you are upgrading from older versions of Grafana please be sure to check your dashboard config and check for warnings.
